### PR TITLE
Plat UI 3141 adding UI tests

### PR DESCRIPTION
**Pull request by goodvibes**
https://jira.tools.tax.service.gov.uk/browse/PLATUI-3141

Created tests to match functionality of those being removed from help-frontend

### CODE REVIEW

File: src/test/scala/uk/gov/hmrc/ui/pages/BasePage.scala [hunk 1]
```diff
@@ -16,6 +16,27 @@
...
-trait BasePage extends PageObject {}
+trait BasePage extends PageObject {
+
+  protected val url: String
+
+  def webDriver: WebDriver = Driver.instance
+
+  def goTo(): Unit = get(url)
+
+  def pageTitle(): String = webDriver.getTitle
```
Comment (vivrichards600) [inline src/test/scala/uk/gov/hmrc/ui/pages/BasePage.scala]: By using the ui-test-runner PageObject.scala, there are a few helpers which could simplify some of the code. This is absolutely optional and down to teams to decide if they want to use it or not :-) 

For example, here you could consider:

`def pageTitle(): String = getTitle`

File: src/test/scala/uk/gov/hmrc/ui/pages/CookiesPage.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+package uk.gov.hmrc.ui.pages
+
+import org.openqa.selenium.By
+import uk.gov.hmrc.configuration.TestEnvironment
+
+sealed trait CookiesPage extends BasePage {
+
+  override protected val url: String = TestEnvironment.url("help-frontend") + "/cookie-details"
+
+  def cookiesInfoText(): String = webDriver.findElement(By.id("cookies-info")).getText
```
Comment (vivrichards600) [inline src/test/scala/uk/gov/hmrc/ui/pages/CookiesPage.scala]: Optional, you could instead using the helper use:

`  def cookiesInfoText(): String = getText(By.id("cookies-info"))`

File: src/test/scala/uk/gov/hmrc/ui/pages/CookiesPage.scala [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  def otherLanguage: String
+
+  def switchLanguage(): Unit =
+    webDriver.findElement(By.partialLinkText(otherLanguage)).click()
```
Comment (vivrichards600) [inline src/test/scala/uk/gov/hmrc/ui/pages/CookiesPage.scala]: Optional: Could also use:

`  def switchLanguage(): Unit =
    click(By.partialLinkText(otherLanguage))`

File: src/test/scala/uk/gov/hmrc/ui/pages/OnlineServicesTermsPage.scala [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    val subHeadingElements = webDriver.findElements(By.className("govuk-heading-l")).asScala
+    subHeadingElements.toSeq.map(_.getText)
+  }
+
+  def languageOfPage(): String = webDriver
+    .findElement(By.tagName("html"))
+    .getAttribute("lang")
+
+  def switchLanguageToEnglish(): Unit =
+    webDriver.findElement(By.partialLinkText("English")).click()
```
Comment (vivrichards600) [inline src/test/scala/uk/gov/hmrc/ui/pages/OnlineServicesTermsPage.scala]: Optional, could also use:
`  def switchLanguageToEnglish(): Unit =
    click(By.partialLinkText("English"))`

File: src/test/scala/uk/gov/hmrc/ui/pages/OnlineServicesTermsPage.scala [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+  def switchLanguageToWelsh(): Unit =
+    webDriver.findElement(By.partialLinkText("Cymraeg")).click()
```
Comment (vivrichards600) [inline src/test/scala/uk/gov/hmrc/ui/pages/OnlineServicesTermsPage.scala]: Optional, could also use:
`  def switchLanguageToWelsh(): Unit =
    click(By.partialLinkText("Cymraeg"))`

File: src/test/scala/uk/gov/hmrc/ui/pages/TermsAndConditionsPage.scala [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+package uk.gov.hmrc.ui.pages
+
+import org.openqa.selenium.By
+import uk.gov.hmrc.configuration.TestEnvironment
+
+object TermsAndConditionsPage extends BasePage {
+
+  override protected val url: String = TestEnvironment.url("help-frontend") + "/terms-and-conditions"
+
+  def disclaimerText(): String = webDriver.findElement(By.id("disclaimer")).getText
```
Comment (vivrichards600) [inline src/test/scala/uk/gov/hmrc/ui/pages/TermsAndConditionsPage.scala]: Optional, could also use:
`def disclaimerText(): String = getText(By.id("disclaimer"))`

**Review by vivrichards600 — COMMENTED** [review]
Added some comments, you don't need to address these if you don't want to use the helpers provided by ui-test-runner. We don't force teams to use these helpers.

**Review by vivrichards600 — CHANGES_REQUESTED** [review]
The first line of the readme still says:
`
This is the template README. Please update this with project specific content.
`

File: src/test/scala/uk/gov/hmrc/ui/pages/BasePage.scala (hunk 1 above)
Comment (goodvibes) [reply src/test/scala/uk/gov/hmrc/ui/pages/BasePage.scala]: Done

File: src/test/scala/uk/gov/hmrc/ui/pages/CookiesPage.scala (hunk 2 above)
Comment (goodvibes) [reply src/test/scala/uk/gov/hmrc/ui/pages/CookiesPage.scala]: done

File: src/test/scala/uk/gov/hmrc/ui/pages/CookiesPage.scala (hunk 3 above)
Comment (goodvibes) [reply src/test/scala/uk/gov/hmrc/ui/pages/CookiesPage.scala]: done

File: src/test/scala/uk/gov/hmrc/ui/pages/OnlineServicesTermsPage.scala (hunk 4 above)
Comment (goodvibes) [reply src/test/scala/uk/gov/hmrc/ui/pages/OnlineServicesTermsPage.scala]: done

**Review by vivrichards600 — APPROVED** [review]

**Pull request merged**